flocker/node/script.py

```diff
@@ -113,32 +109,42 @@ def agent_config_from_file(path):
                     },
                     "port": {"type": "integer"},
                 }
+            },
+            "dataset": {
+                "type": "object",
+                "oneOf": [
+                    {
+                        "required": ["backend"],
+                        "properties": {
+                            "backend": {
+                                "type": "string",
+                                "pattern": "zfs",
+                            },
+                            "zfs-pool": {
```

Comment:
I don't think we need to prefix options here. It doesn't make sense for a configuration file to have configuration for multiple backends at once in it.